src/gluonts/torch/model/gnninf/networks/model.py

```diff
@@ -0,0 +1,234 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# If you use this code in your work please cite:
+# Multivariate Time Series Forecasting with Latent Graph Inference
+# (https://arxiv.org/abs/2203.03423)
+
+import torch
+from torch import nn
+from . import gnns
+from . import utils
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+class GNNInfNetwork(torch.nn.Module):
+    """
+    Input and Output tensor will have the shape:
+    (batch_size, num_nodes, seq_len, dimensions)
+    :param in_channels: number of channels/dimensions in the input sequence
+    (this value is usually set to 1)
```

Comment:
Better to use [numpy-style docstrings](https://numpydoc.readthedocs.io/en/latest/format.html#style-guide), see for example https://github.com/awslabs/gluon-ts/blob/ae4cfdef539e49f93a87034aa2f2bec194c4b7d8/src/gluonts/model/deepar/_estimator.py#L69-L89